pygmt/src/subplot.py

```diff
@@ -0,0 +1,222 @@
+"""
+subplot - Manage modern mode figure subplot configuration and selection.
+"""
+import contextlib
+
+from pygmt.clib import Session
+from pygmt.helpers import (
+    build_arg_string,
+    fmt_docstring,
+    is_nonstr_iter,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@contextlib.contextmanager
+@use_alias(
+    Ff="figsize",
+    Fs="subsize",
+    A="autolabel",
+    B="frame",
+    C="clearance",
+    J="projecton",
+    M="margins",
+    R="region",
+    SC="sharex",
+    SR="sharey",
+    T="title",
+    V="verbose",
+    X="xshift",
+    Y="yshift",
+)
+@kwargs_to_strings(Ff="sequence", Fs="sequence", M="sequence", R="sequence")
+def subplot(self, nrows=1, ncols=1, **kwargs):
+    r"""
+    Create multi-panel subplot figures.
+
+    This function is used to split the current figure into a rectangular layout
+    of subplots that each may contain a single self-contained figure. Begin by
+    defining the layout of the entire multi-panel illustration. Several
+    arguments are available to specify the systematic layout, labeling,
+    dimensions, and more for the subplots.
+
+    Full option list at :gmt-docs:`subplot.html#synopsis-begin-mode`
+
+    {aliases}
+
+    Parameters
+    ----------
+    nrows : int
+        Number of vertical rows of the subplot grid.
+    ncols : int
+        Number of horizontal columns of the subplot grid.
+    figsize : tuple
+        Specify the final figure dimensions as ``(width, height)``.
+    subsize : tuple
+        Specify the dimensions of each subplot directly as ``(width, height)``.
```

Comment:
Add a check to make sure that users don't give both `figsize` and `subsize`?

Reply:
Done at c475861383423bfa312e2bd625a4ea61682c14be.